### ISSUE TITLE
Feedback: Capacity units

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/quartz-frontend#472]

**Comment by peterdudfield** [comment]
Also having a hover over the GSP, would perhaps show what units its in

**Comment by akshayw1** [comment]
Could you pls assign this to me?

**Comment by peterdudfield** [comment]
@braddf just so you are aware, @akshayw1 might try this

**Comment by akshayw1** [comment]
Can you breif more over the issue, so that it could be clear to me @peterdudfield

**Comment by akshayw1** [comment]
Is there a contributor's document available for setting up the project locally? @peterdudfield

**Comment by akshayw1** [comment]
I have a couple of doubts regarding the issue. The repository contains two apps, Nowcasting and Quartz. Could you clarify which app this issue is related to? Additionally, could you specify the exact route involved? This will help me address and resolve the issue more quickly.

**Comment by peterdudfield** [comment]
> I have a couple of doubts regarding the issue. The repository contains two apps, Nowcasting and Quartz. Could you clarify which app this issue is related to? Additionally, could you specify the exact route involved? This will help me address and resolve the issue more quickly.

nowcasting

**Comment by akshayw1** [comment]
Is there a communication channel available for OFX support? I’ve successfully set up both projects locally, but I’m encountering an issue with unauthorized access. Any guidance or support would be greatly appreciated. 

I have setup all from auth0 and added https://api.quartz.solar/ IN API's too, Still getting unauthorised

Here’s a screenshot of the error for reference:  
![image](https://github.com/user-attachments/assets/78214ecd-2fd3-448f-b403-147e745dc58e)

![image](https://github.com/user-attachments/assets/72c5d1d4-4112-4ca5-a9dc-560e0238276e)

**Comment by peterdudfield** [comment]
The best way is on github discussions. I think this might be an issue that @braddf is aware off.

**Comment by peterdudfield** [comment]
But for a quick comment, are you passing a bearer token to the API?

**Comment by akshayw1** [comment]
@peterdudfield  ofc yes, I have pass the bearer token, in request headers
![image](https://github.com/user-attachments/assets/d9400ed0-26be-4b2b-8759-442a3e27733b)

**Comment by peterdudfield** [comment]
Your are using `api-dev`, did the bearer token get generated for that environment? Actually how did you make the bearer token?

**Comment by akshayw1** [comment]
The bearer token was generated specifically for the **api-dev** environment using the client credentials provided for this setup. The process involved authenticating via the OAuth endpoint with the correct client ID and secret for **api-dev**. 

If there’s a possibility the token is mismatched or not valid for this environment, could you confirm if additional steps or configurations are required for generating a valid bearer token for **api-dev**?

**Comment by peterdudfield** [comment]
> The bearer token was generated specifically for the **api-dev** environment using the client credentials provided for this setup. The process involved authenticating via the OAuth endpoint with the correct client ID and secret for **api-dev**.
> 
> If there’s a possibility the token is mismatched or not valid for this environment, could you confirm if additional steps or configurations are required for generating a valid bearer token for **api-dev**?

No, that sounds about right. Do you get any more details in the error?

**Comment by akshayw1** [comment]
Can you confirm me API_URL that is used, shoudl I try for https://api.quartz.solar/

**Comment by peterdudfield** [comment]
https://api-dev.quartz.solar/swagger looks good, thats the development env. One way you could do it is, go on to app.quartz.solar, use the inspect feature to get your bearer token. And use the bearer token for api.quartz.solar. Just an idea, and I dont know how feasible it is

**Comment by akshayw1** [comment]
Pls let me know,If possible, could we connect over a Meet session or another platform to review the configuration together? I believe a quick discussion could help identify and resolve the issue efficiently.
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/6d35064b-7cf2-4836-9e36-e9a2aed2b2e7" />

**Comment by peterdudfield** [comment]
Sorry. I think we are both away until early January. I'm sure we can pick it up then

**Comment by akshayw1** [comment]
Sure @peterdudfield , Thanks I would try and wait til, about me
I'm Akshay Waghmare, an undergrad at IIIT Allahabad, and I'm excited about contributing to Open Climate Fix for GSoC 2025. I recently got my first PR merged into the PV Live consumer project and would love to contribute further. I have experience in JavaScript, Python, ML, and web technologies, and I'm eager to work on projects involving real-time data analysis and AI solutions for climate challenges. Any guidance on how I can actively contribute would be appreciated!

**Comment by peterdudfield** [comment]
Did you have any luck @akshayw1 ?

**Comment by akshayw1** [comment]
After exploring it briefly, I shifted my focus to other things since I didn’t hear back from you. I was eagerly looking forward to your response and am now ready to dive back in! @peterdudfield

**Comment by peterdudfield** [comment]
So where did you get too? 
You said you managed to make a create a brearer token? Did this work for api.quartz.solar or api-dev.quartz.solar? And did you pass that bearer token through with the code in this repo

**Comment by akshayw1** [comment]
I believe all the necessary configurations were set up correctly, including the bearer token, OAuth authentication, and adding the domain. The issue likely stemmed from something else that needs to be identified and addressed.

**Comment by akshayw1** [comment]
@peterdudfield any update

**Comment by peterdudfield** [comment]
I think @braddf is working on something

**Comment by braddf** [comment]
@akshayw1 Sorry I missed these comments, Github overload! 

At present, as per the `nowcasting-app` README, the only way for OS contributors to run the UI locally is with a local API instance running with fake data. The [uk-pv-national-gsp-api](https://github.com/openclimatefix/uk-pv-national-gsp-api) repo has instructions on how to get this up and running and generating the fake data.

We're not working on any other way to auth this, and we're actually going to be gating off repos like this one from contribution, mostly because it's too tricky for our small team to support useful folks like yourself with setting up and running environments without access to our systems. 

I just came back here on an overhaul of our Roadmap and saw your comments; FYI we're actually pivoting this issue into another, so not worth making these specific changes.
For now, there are some small issues I can add `good-first-issue` tags to if you are still interested in contributing to this repo, let me know if so, but no worries if not. 🙏

**Comment by suvanbanerjee** [comment]
GSP's 
<img width="1269" height="894" alt="Image" src="https://github.com/user-attachments/assets/48a48881-e645-4e55-9911-54eb0a9dc00e" />

DNO's 
<img width="1269" height="894" alt="Image" src="https://github.com/user-attachments/assets/261bee0c-874e-4906-9a10-27057283452e" />

PS -  it is normalized by max cap density -> 1 (GSP), 0.2 (DNO) 

This is how it looks -- good for a PR? 
cc @braddf @peterdudfield

**Comment by suvanbanerjee** [comment]
Preview URL - https://nowcasting-3640hof1j-openclimatefix.vercel.app/
